### PR TITLE
Remove underscores from docs page names

### PR DESCRIPTION
**Pull request by stichbury**
## Description
This PR changes names as #313 describes.

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/API-reference/manager.md
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/API-reference/manager.md:3]: I think this is the reason why the docs build is currently unsuccessful :) 

```suggestion
::: vizro.managers._data_manager
```

File: vizro-core/docs/pages/user-guides/actions.md
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/actions.md:20]: I think this actually needs to stay as previously as otherwise the note is not rendered correctly:

Screenshot 2024-02-19 at 19 48 07

File: vizro-ai/docs/pages/user-guides/api-setup.md
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/user-guides/api-setup.md:29]: Revert formatting here as otherwise not displayed correctly

File: vizro-ai/docs/pages/user-guides/api-setup.md
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/user-guides/api-setup.md:76]: Revert formatting here as otherwise not displayed correctly

**Review by huong-li-nguyen — CHANGES_REQUESTED** [review]
Hey @stichbury - looks good to me in general! 🚀 Thanks for fixing it!

I've commented on the code line which currently prevents the docs to render properly.

I also think you need to revert some of the formatting changes on the [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/), because otherwise it doesn't get displayed properly due to mkdocs expecting that kind of input. I've commented on a few but would be great if you can double-check throughout the PR! :) 

So the formatting needs to stay like this:
Screenshot 2024-02-19 at 19 53 20

After that, let me know if you need help on the dev set-up. The current failing CI tests indicate that a changelog file is missing in vizro-ai and vizro-core. Would be great to see if the error messages are sufficient to solve it on your own, otherwise let me know and I can help out! :)

**Comment by antonymilne** [comment]
@stichbury in case you didn't see it, we have RTD builds done on every PR so you can check rendering easily.
image

Also would it make sense if we add to to CODEOWNERS for docs?

**Comment by stichbury** [comment]
Oh, whoops, sorry! I didn't mean this to be ready for review yet, it should have gone into draft. Apologies, but thanks for the feedback as that'll make it easier for me to find and fix those issues in the morning. I didn't change any formatting so am not sure how that's happened but will roll it back so it's just file renames in this PR 🤦 

Yes to CODEOWNERS too please!

File: vizro-ai/docs/pages/user-guides/api-setup.md
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/api-setup.md:17]: This is so strange, I didn't change this!

Comment (huong-li-nguyen) [reply vizro-ai/docs/pages/user-guides/api-setup.md:17]: Ahhhh, I know why 😅 Our automatic pre-commit auto-fix did that there: https://github.com/mckinsey/vizro/pull/320/commits/80db16e82c7870ec9d366ae0a3b85d93747782f6


The reason for that was that we ignore these folders here, but because you've changed the names, they didn't get ignored anymore. So you would also have to update this file:

https://github.com/mckinsey/vizro/blob/main/.prettierignore

Comment (stichbury) [reply vizro-ai/docs/pages/user-guides/api-setup.md:17]: Oh, that makes sense, thanks for diagnosing that for me! I shall fix that....and then go through and fix any broken links I've made in the content. Should be ready for a second review today!

Comment (huong-li-nguyen) [reply vizro-ai/docs/pages/user-guides/api-setup.md:17]: Let me know if you need help with reverting the auto-fix commit and/or the changelog files :)

**Pull request closed without merging**
